### DISCUSSION TITLE
How to model boolean features (no value) and is there a way to set groups of features ?

### DISCUSSION THREAD

**Discussion by ghost** [Flagsmith/flagsmith#1871] [Q&A] [answered]
Hi,
I have a question which might be general to FF design, please feel free to close.

I'm trying to use *Flagsmith* to set different billing plans for classes of users. Can you please give me your opinion on this ? as FF are generally used to have different behaviour for different environments (local, prod etc). For me I'm using features for teams of users like this:

```
{ 
    feat1_freemium : true, 
    feat1_premium : false, 
    feat1_business: false, 
    feat2_parc_freemium : true, 
    feat2_parc_premium : false, 
    feat2_parc_business : false, 
    ... 
}  
```
Is this a good way to model this ? Also setting boolean as a value doesn't seem to be necessary in this particular case, as a feature by definition is either enabled (true) or not (false). 

What I aimed for and thought naively there is a way to do, is to set default features for not logged in users (default features), but for logged in users (custom features), I would defined their own set of features manually. A concept of teams of users would be great (for my own case).

Visitors:
```
{ 
    feat1_freemium : true, 
    feat1_premium : false, 
    feat1_business: false, 
    feat2_parc_freemium : true, 
    feat2_parc_premium : false, 
    feat2_parc_business : false, 
    ... 
}  
```

Logged in users (customers)
Visitors:
```
// user of class premium
{ 
    feat1_freemium : true, 
    feat1_premium : true, 
    feat1_business: false, 
    feat2_parc_freemium : true, 
    feat2_parc_premium : true, 
    feat2_parc_business : false, 
    ... 
}  
```

```
// user of class business
{ 
    feat1_freemium : true, 
    feat1_premium : true, 
    feat1_business: true, 
    feat2_parc_freemium : true, 
    feat2_parc_premium : true, 
    feat2_parc_business : true, 
    ... 
}  
```
**Edit**: it seems I can use identities, and map users to a set of groups in app level somehow, this might be an option.

Thanks a lot :)

gz#361

**Comment by dabeeeenster** [comment, marked as answer]
It might be better to use Flag Values as opposed to booleans, then check for that value in your code when enabling feateures. So have a value for each feature, which is the minimum plan level that it is available at. E.g.

``` 
feat1: premium,
feat2: business,
feat3: freemium
```

Then set an Identity Trait for the plan the user is on.

**Reply by ghost** [reply]
indeed, this looks a better design. Thanks!
I would repeat features for plans though, as i'm 90% sure features are not incremental vs plans. Like one feature can be active on a plan but not the other and vice versa. (arbitrary rules)

**Comment by dabeeeenster** [comment]
Store a JSON list as a value of allowed plans.

**Reply by ghost** [reply]
I didn't get the point, you mean to avoid setting all manually each time ? to use [direct API](https://docs.flagsmith.com/clients/rest) ?

**Reply by dabeeeenster** [reply]
Have you read through https://docs.flagsmith.com/basic-features/managing-identities and  https://docs.flagsmith.com/basic-features/managing-segments ? 

The most common approach is to set the plan that the user has as a Trait and then use that Trait to drive Segments that override a feature value.
